staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -53,10 +53,31 @@ var (
 		},
 		[]string{"resource"},
 	)
+
+	watcherInputBufferSize = prometheus.NewHistogramVec(
```

Comment:
> This PR improves the process of dispatch an event, and add two metrics to observe the efficiency of watchCache.

interesting. the pull tries to optimize performance but plumbs more metrics. should at least split it.

Reply:
+1 on the multiple commits, there are definitely several things going on here.